gnc/sub8_perception/nodes/follow_orange_pipes.py

```diff
@@ -116,181 +138,148 @@ def find_pipe_new(self, img):
         cv2.drawContours(mask, [box], 0, 255, -1)
         rect_area = cv2.contourArea(box)
 
-        center, angle_rad, [max_eigv, min_eigv] = self.get_pose(mask)
+        center, angle_rad, [max_eigv, min_eigv] = self.get_2d_pose(mask)
+        cv2.line(self.last_draw_image, tuple(np.int0(center)), tuple(np.int0(center + (2 * max_eigv))), (0, 255, 30), 2)
+        cv2.line(self.last_draw_image, tuple(np.int0(center)), tuple(np.int0(center + (2 * min_eigv))), (0, 30, 255), 2)
 
         # Check if the box is too big or small.
-        xy_position, height = self.occ_grid.get_tf(timestamp=last_image_timestamp)
-        expected_area = self.occ_grid.calculate_marker_area(height)
+        xy_position, height = self.get_tf(timestamp=last_image_timestamp)
+        expected_area = self.calculate_marker_area(height)
         if expected_area * .3 < rect_area < expected_area * 2:
             cv2.drawContours(self.last_draw_image, [box], 0, (255, 255, 255), -1)
         else:
             angle_rad = 0
             max_eigv = np.array([0, -20])
             min_eigv = np.array([-20, 0])
             #cv2.drawContours(self.last_draw_image, [box], 0, (255, 0, 30), -1)
-            print "Size out of bounds!"
+            rospy.logwarn("MARKER - Size out of bounds!")
 
-        cv2.line(self.last_draw_image, tuple(np.int0(center)), tuple(np.int0(center + (2 * max_eigv))), (0, 255, 30), 2)
-        cv2.line(self.last_draw_image, tuple(np.int0(center)), tuple(np.int0(center + (2 * min_eigv))), (0, 30, 255), 2)
+        # Convert to a 3d pose to move the sub to.
+        abs_position = self.transform_px_to_m(center, last_image_timestamp)
```

Comment:
Never mind, I see how you did it. Keep in mind that it will be more and more inaccurate the farther you are from having an orthogonal view of the strip.